src/main/java/sirius/kernel/commons/Files.java

```diff
@@ -146,6 +159,34 @@ public static String getFilenameWithoutExtension(@Nullable String path) {
         return Strings.splitAtLast(getFilenameAndExtension(path), ".").getFirst();
     }
 
+    /**
+     * Determines if the given path is hidden.
+     * <p>
+     * A path is considered hidden if it starts with a dot.
+     *
+     * @param path the path to check
+     * @return <tt>true</tt> if the path is hidden, <tt>false</tt> otherwise
+     */
+    public static boolean isHidden(@Nullable String path) {
+        return path != null && path.startsWith(".");
+    }
+
+    /**
+     * Determines if the given path is a metadata file.
+     * <p>
+     * A metadata file is a file which is not part of the actual content but rather contains metadata or is used by
+     * the operating system or other tools.
+     *
+     * @param path the path to check
+     * @return <tt>true</tt> if the path is a metadata file that is listed in the METADATA_FILES list, <tt>false</tt> otherwise
+     */
+    public static boolean isMetadata(@Nullable String path) {
+        if (Strings.isEmpty(path)) {
+            return false;
+        }
+        return METADATA_FILES.stream().anyMatch(path::contains);
```

Comment:
**Required**: sure we want to use contains against the list? This would discard real files with an extension `.ini` which are not necessarily metadata.
I think would be useful to get a hand of a few Windows-generated ZIPS and look what is inside. For macOS is clear and we can check with `.startsWith`

Reply:
Or seeing Sacha's comment, maybe we need a RegEx match to solve that properly, or have at least 2 pattern lists, to check with startsWith and endsWith. It all depends on how these actually appear in the files.

Reply:
Yes, if more than one pattern exist, we can directly go with a list of patterns, which gives us way more flexibility.